### PR TITLE
UI: Service Accounts

### PR DESCRIPTION
**Pull request by dvaldivia**
Screen Shot 2020-02-05 at 3 00 32 PM
Screen Shot 2020-02-05 at 3 00 27 PM
Screen Shot 2020-02-05 at 3 00 20 PM
Screen Shot 2020-02-05 at 3 00 17 PM

### CODE REVIEW

**Comment by cesnietor** [comment]
Service account delete message is showing odd message and typo.
Screen Shot 2020-02-05 at 5 45 29 PM

**Comment by cesnietor** [comment]
You can't un check 
Screen Shot 2020-02-05 at 5 48 18 PM

**Comment by cesnietor** [comment]
Same if name is too long.
Please do this for users if not implemented yet.

Screen Shot 2020-02-05 at 5 51 28 PM

**Comment by dvaldivia** [comment]
Screen Shot 2020-02-06 at 10 47 09 AM
@cesnietor fixed

File: portal-ui/src/screens/Console/ServiceAccounts/DeleteServiceAccount.tsx [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+                        <b className={classes.wrapText}>{selectedServiceAccount.name}</b>?
+                        {deleteError !== "" && (
+                            <React.Fragment>
+                                <br />
+                                <Typography
+                                    component="p"
+                                    variant="body1"
+                                    className={classes.errorBlock}
+                                >
+                                    {`${deleteError}`}
```
Comment (Alevsk) [inline portal-ui/src/screens/Console/ServiceAccounts/DeleteServiceAccount.tsx]: Please change `{`${deleteError}`}` to `{deleteError}`

File: portal-ui/src/screens/Console/ServiceAccounts/ServiceAccounts.tsx [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+import DeleteServiceAccount from "./DeleteServiceAccount";
+import CredentialsPrompt from "./CredentialsPrompt";
+
+const styles = (theme: Theme) =>
+  createStyles({
+    seeMore: {
+      marginTop: theme.spacing(3)
+    },
+    paper: {
+      // padding: theme.spacing(2),
```
Comment (Alevsk) [inline portal-ui/src/screens/Console/ServiceAccounts/ServiceAccounts.tsx]: remove this line

File: portal-ui/src/screens/Console/ServiceAccounts/ServiceAccounts.tsx [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+    this.setState({ loading: true }, () => {
+      const { page, rowsPerPage } = this.state;
+      const offset = page * rowsPerPage;
+      api
+        .invoke(
+          "GET",
+          `/api/v1/service_accounts?offset=${offset}&limit=${rowsPerPage}`
+        )
+        .then((res: ServiceAccountsList) => {
+          this.setState({
```
Comment (Alevsk) [inline portal-ui/src/screens/Console/ServiceAccounts/ServiceAccounts.tsx:143]: just to consider that setState is async function, dont know if we want to wait til is donde to run the following code

```
          this.setState({ page: newPage }, () => {
              this.fetchRecords();
            });
```

in that case ^ should be added as a callback in this.setState(state, callback function)

**Review by Alevsk — CHANGES_REQUESTED** [review]

File: portal-ui/src/screens/Console/ServiceAccounts/ServiceAccounts.tsx (hunk 2 above)
Comment (dvaldivia) [reply portal-ui/src/screens/Console/ServiceAccounts/ServiceAccounts.tsx]: solved

File: portal-ui/src/screens/Console/ServiceAccounts/ServiceAccounts.tsx (hunk 3 above)
Comment (dvaldivia) [reply portal-ui/src/screens/Console/ServiceAccounts/ServiceAccounts.tsx:143]: In this particular case it's already ok to set the state immediately, in other places in the call, callbacks are passed to set state

**Review by Alevsk — APPROVED** [review]

**Pull request merged**